adyenofficial.php

```diff
@@ -1011,71 +1108,108 @@ public function displayForm()
             'required' => true
         );
 
-        $apiKeyTest = '';
+        $apiKeyLive = '';
 
         try {
-            $apiKeyTest = $this->crypto->decrypt(Configuration::get('ADYEN_APIKEY_TEST'));
+            $apiKeyLive = $this->crypto->decrypt(Configuration::get('ADYEN_APIKEY_LIVE'));
         } catch (\Adyen\PrestaShop\exception\GenericLoggedException $e) {
             $this->logger->error(
-                'For configuration "ADYEN_APIKEY_TEST" an exception was thrown: ' . $e->getMessage()
+                'For configuration "ADYEN_APIKEY_LIVE" an exception was thrown: ' . $e->getMessage()
             );
         } catch (\Adyen\PrestaShop\exception\MissingDataException $e) {
-            $this->logger->warning('The configuration "ADYEN_APIKEY_TEST" has no value set.');
+            $this->logger->warning('The configuration "ADYEN_APIKEY_LIVE" has no value set.');
         }
 
-        $apiKeyTestLastDigits = Tools::substr($apiKeyTest, -4);
+        $apiKeyLiveLastDigits = Tools::substr($apiKeyLive, -4);
 
-        $fields_form[0]['form']['input'][] = array(
+        $fields_form[1]['form']['input'][] = array(
             'type' => 'password',
             // phpcs:ignore Generic.Files.LineLength.TooLong
-            'label' => '<a target="_blank" href="https://docs.adyen.com/plugins/prestashop#generate-an-api-key">API key for Test</a>',
-            'name' => 'ADYEN_APIKEY_TEST',
-            'desc' => $apiKeyTestLastDigits ? $this->l('Saved key ends in: ') . $apiKeyTestLastDigits :
-                $this->l('Please fill your API key for Test'),
-            'class' => $apiKeyTestLastDigits ? 'adyen-input-green' : '',
+            'label' => '<a target="_blank" href="https://docs.adyen.com/plugins/prestashop#generate-an-api-key">API key for Live</a>',
+            'name' => 'ADYEN_APIKEY_LIVE',
+            'desc' => $apiKeyLiveLastDigits ? $this->l('Saved key ends in: ') . $apiKeyLiveLastDigits :
+                $this->l('Please fill your API key for Live'),
+            'class' => $apiKeyLiveLastDigits ? 'adyen-input-green' : '',
```

Comment:
Since we got some feedback for the green border color maybe we can change this to password and if the value is ***** then we just act like nothing is submitted. Therefore we can get rid of the green coloring, we still don't pass back the unencrypted API key and it's maybe easier for the merchants to understand the UI.

Reply:
https://github.com/Adyen/adyen-prestashop/pull/203/commits/e46540447f73d84ddf9a31e84142a3e6caf8b815